modin/core/dataframe/pandas/dataframe/utils.py

```diff
@@ -0,0 +1,246 @@
+    ----------
+    modin_frame : PandasDataframe
+        A reference to the frame calling these sort functions.
+    columns : list[str]
+        The list of column names to sort by.
+    method : str
+        The method to use for picking quantiles.
+    ascending : list[bool] or bool
+        The ascending flag (or a list of ascending flags for each column).
+    **kwargs : dict
+        Additional keyword arguments.
+
+    Returns
+    -------
+    dict :
+        A dictionary containing the functions to pick quantiles, pick overall quantiles, and split
+        partitions for sorting.
+    """
+
+    def sample_fn(partition, A=100, k=0.05, q=0.1):
+        return get_partition_quantiles_for_sort(
+            partition, columns, A=A, k=k, q=q, method=method
+        )
+
+    def pivot_fn(samples):
+        return pick_pivots_from_quantiles_for_sort(
+            modin_frame, samples, columns, method
+        )
+
+    def split_fn(partition, pivots):
+        return split_partitions_using_pivots_for_sort(
+            modin_frame, partition, columns, pivots, ascending, **kwargs
+        )
+
+    return {
+        "sample_function": sample_fn,
+        "pivot_function": pivot_fn,
+        "split_function": split_fn,
+    }
+
+
+def get_partition_quantiles_for_sort(
+    df: pandas.DataFrame,
+    columns: list,
+    A: int = 100,
+    k: float = 0.05,
+    q: float = 0.1,
+    method: str = "linear",
+) -> "tuple[pandas.DataFrame, np.ndarray]":
+    """
+    Pick quantiles over the given partition.
+
+    This function is applied to each row-axis partition in parallel to select quantiles
+    from each. It samples using the following algorithm:
+        * If there are <= A (100) rows in the dataframe, we pick quantiles over the entire dataframe.
+        * If there are A (100) < # of rows <= (A * (1 - k))/(q - k) (1900), we pick quantiles over the first A (100) rows, plus a sample
+        of k (5%) of the remaining rows.
+        * If there are > (A * (1 - k))/(q - k) (1900) rows, we pick quantiles over a sample of q (10%) of all of the rows.
+
+    These numbers are a heuristic. They were picked such that the size of the sample
+    in scenario 2 scales well. In other words, we picked A, k, and q such that:
+        A + (len(df) - A)*k = q*len(df)
+    Where q is the proportion we sample in scenario 3, k is the proportion we sample
+    of remaining rows in scenario 2, and A is the threshold below which we just return
+    the entire dataframe, instead of sampling (scenario 1).
+    q = 0.1 and k = 0.05 were picked such that this formula holds for A = 100.
+
+    Parameters
+    ----------
+    df : pandas.Dataframe
+        The dataframe to pick quantiles from.
+    columns : list[str]
+        The columns to pick quantiles from. Only the first column in the list will be used.
+    A : int, default: 100
+        A heuristic that defines "small" dataframes.
+    k : float, default: 0.05
+        A heuristic used for sampling.
```

Comment:
```suggestion
        If A < len(df) <= (A * (1 - k))/(q - k), select A + k * (len(df) - A) elements
```

not sure that I got the math right, but the point is we can give a more precise description.